File: examples/core/spark-config/src/main/scala/my/spark/core/SparkConfig.scala
package my.spark.core

import org.apache.spark.SparkConf
import org.apache.spark.SparkContext

object SparkConfig {
  def main(args: Array[String]) {

    if (args.length > 1) {
      System.err.println("Usage: SparkConfig <appName>")
      System.exit(1)
    }

    if (args.length == 1) {
      val conf = new SparkConf().setAppName(args(0))
      val sc = new SparkContext(conf)
      println("Application Name: " + sc.appName)
      sc.stop()
    }
    
    else {
      val sc = new SparkContext()
      println("Application Name: " + sc.appName)
      sc.stop()
    }
  }
}

File: examples/core/spark-config/src/main/python/SparkConfig.py
import sys
from pyspark import SparkConf
from pyspark import SparkContext

if __name__ == "__main__":
  if len(sys.argv) > 2:
    print >> sys.stderr, "Usage: SparkConfig <appName>"
    exit(-1)

  if len(sys.argv) == 2:
    conf = SparkConf().setAppName(sys.argv[1])
    sc = SparkContext(conf=conf)
    print("Application Name: " + sc.appName)
    sc.stop()

  else:
    sc = SparkContext()
    print("Application Name: " + sc.appName)
    sc.stop()
